SELinux in enforcing mode breaks nested gVisor container

# Problem

While debugging a Dangerzone 0.7.0 bug report on a Fedora 40 system (#871), we found out that SELinux in enforcing mode breaks the **inner** gVisor container that runs within the **outer** Podman container.

## Original error

```
$ dangerzone-cli ~/Downloads/sample.pdf
[...]
[DEBUG] Marking doc qyU8c4 as 'converting'
[INFO] > /usr/bin/podman run --log-driver none --security-opt no-new-privileges --cap-drop all --cap-add SYS_CHROOT --network=none -u dangerzone --rm -i --name dangerzone-doc-to-pixels-qyU8c4 --userns nomap dangerzone.rocks/dangerzone /usr/bin/python3 -m dangerzone.conversion.doc_to_pixels
[ERROR] [doc qyU8c4] 0% Unspecified error
[DEBUG] Marking doc qyU8c4 as 'failed'
```

## Output from Podman container with gVisor debug logging enabled

```
$ /usr/bin/podman run -e RUNSC_DEBUG=1 --log-driver none --security-opt no-new-privileges --cap-drop all --cap-add SYS_CHROOT --network=none -u dangerzone --rm -i --name dangerzone-doc-to-pixels-qyU8c4 --userns nomap dangerzone.rocks/dangerzone /usr/bin/python3 -m dangerzone.conversion.doc_to_pixels

8< ... snip ... >8

I0725 14:19:04.811776      20 main.go:197] **************** gVisor ****************
I0725 14:19:04.812445      20 boot.go:258] Setting product_name: "Standard PC (Q35 + ICH9, 2009)"
W0725 14:19:04.813081      20 specutils.go:129] noNewPrivileges ignored. PR_SET_NO_NEW_PRIVS is assumed to always be set.
I0725 14:19:04.813105      20 chroot.go:91] Setting up sandbox chroot in "/tmp"
W0725 14:19:04.813312      20 chroot.go:108] Failed to copy /etc/localtime: open /etc/localtime: no such file or directory. UTC timezone will be used.
I0725 14:19:04.813335      20 chroot.go:36] Mounting "/proc" at "/tmp/proc"
W0725 14:19:04.813394      20 util.go:64] FATAL ERROR: error setting up chroot: error remounting chroot in read-only: permission denied
error setting up chroot: error remounting chroot in read-only: permission denied
D0725 14:19:04.814640       7 sandbox.go:1278] Destroying sandbox "dangerzone"
D0725 14:19:04.814726       7 sandbox.go:1287] Killing sandbox "dangerzone"
D0725 14:19:04.814809       7 container.go:776] Destroy container, cid: dangerzone
D0725 14:19:04.814847       7 container.go:1087] Killing gofer for container, cid: dangerzone, PID: 15
W0725 14:19:04.814928       7 util.go:64] FATAL ERROR: running container: creating container: cannot create sandbox: cannot read client sync file: waiting for sandbox to start: EOF
running container: creating container: cannot create sandbox: cannot read client sync file: waiting for sandbox to start: EOF
W0725 14:19:04.815004       7 main.go:227] Failure to execute command, err: 1
gVisor quit with exit code: 128
```

## SELinux denial taken from audit logs

> Jul 25 17:51:56 10.0.2.15 audit[8164]: AVC avc:  denied  { mounton } for  pid=8164 comm="exe" path="/proc/fs" dev="proc" ino=4026531847 scontext=system_u:system_r:container_t:s0:c139,c177 tcontext=system_u:object_r:proc_t:s0 tclass=dir permissive=0

# Workarounds

We are currently aware of the following two workarounds:
1. Set SELinux to **permissive** mode: `sudo setenforce Permissive`
2. Disable SELinux labeling for the spawned Podman container with `--security-opt label=disable` (requires changes to Dangerzone code).

--- a/dangerzone/isolation_provider/container.py
+++ b/dangerzone/isolation_provider/container.py
@@ -106,6 +106,9 @@ class Container(IsolationProvider):
           running gVisor.
         * Do not allow access to the network stack.
         * Run the container as the unprivileged `dangerzone` user.
+        * Set the `container_engine_t` SELinux label, which allows gVisor to work on
+          SELinux-enforcing systems
+          (see https://github.com/freedomofpress/dangerzone/issues/880).
 
         For Podman specifically, where applicable, we also add the following:
         * Do not log the container's output.
@@ -141,6 +144,7 @@ class Container(IsolationProvider):
 
         security_args += ["--cap-drop", "all"]
         security_args += ["--cap-add", "SYS_CHROOT"]
+        security_args += ["--security-opt", "label=type:container_engine_t"]
 
         security_args += ["--network=none"]
         security_args += ["-u", "dangerzone"]

